File: FloatItWatch/src/main/java/net/geekstools/floatshort/PRO/Utils/RemoteTask/CreateFloatingShortcuts.java
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 4/25/20 11:54 AM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */

package net.geekstools.floatshort.PRO.Utils.RemoteTask;

import android.app.Activity;
import android.os.Bundle;

import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass;

public class CreateFloatingShortcuts extends Activity {

    FunctionsClass functionsClass;

    String packageName, className;

    @Override
    protected void onCreate(Bundle Saved) {
        super.onCreate(Saved);
        functionsClass = new FunctionsClass(getApplicationContext());

        packageName = getIntent().getStringExtra("PackageName");
        className = getIntent().getStringExtra("ClassName");

        functionsClass.runUnlimitedShortcutsServiceHIS(packageName, className);

        finish();
    }
}

File: FloatItWatch/src/main/java/net/geekstools/floatshort/PRO/Utils/RemoteTask/CreateFloatingShortcuts.kt
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 4/26/20 7:46 AM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */
package net.geekstools.floatshort.PRO.Utils.RemoteTask

import android.app.Activity
import android.os.Bundle
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass

class CreateFloatingShortcuts : Activity() {

    override fun onCreate(Saved: Bundle?) {
        super.onCreate(Saved)

        if (intent.hasExtra("PackageName")
                && intent.hasExtra("ClassName")) {

            val functionsClass = FunctionsClass(applicationContext)

            val packageName = intent.getStringExtra("PackageName")
            val className = intent.getStringExtra("ClassName")

            functionsClass.runUnlimitedShortcutsServiceHIS(packageName, className)
        }

        this@CreateFloatingShortcuts.finish()
    }
}